crates/fuel-core/src/database/contracts.rs

```diff
@@ -149,6 +153,43 @@ impl StorageWrite<ContractsRawCode> for Database {
 }
 
 impl Database {
+    pub fn iter_contract_state_configs(
+        &self,
+    ) -> impl Iterator<Item = StorageResult<ContractStateConfig>> + '_ {
```

Comment:
Nit: shouldn't we know this type before compile time? It might be ugly and this might be cleaner, it just feels interesting hiding the type from caller this way.